config/jobs/kubernetes/release/release-config.yaml

```diff
@@ -49,12 +49,14 @@ presubmits:
       path_alias: k8s.io/test-infra
     spec:
       containers:
-      - image: launcher.gcr.io/google/bazel:0.29.1
+      - image: launcher.gcr.io/google/bazel:1.2.1
         command:
-        - ../test-infra/hack/bazel.sh
-        args:
-        - test
-        - //...
+          - /bin/bash
+          - -c
+          - >
+            apt-get update &&
+            apt-get install -y build-essential &&
```

Comment:
what is this about exactly? this smells like a flake / reproducibility nightmare.

Reply:
The first issue was that golangci-lint does not error if it was not able to execute a dedicated linter. which has been fixed in https://github.com/golangci/golangci-lint/pull/890

Its reproducible by:
```bash
> podman run -v $(pwd):/w -w /w -it --entrypoint bash launcher.gcr.io/google/bazel:2.0.0
root@6509f76f5300:/w# bazel test @io_k8s_repo_infra//hack:verify-golangci-lint
...
INFO: Build completed successfully, 217 total actions
root@6509f76f5300:/w# cat bazel-testlogs/external/io_k8s_repo_infra/hack/verify-golangci-lint/test.log
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from @io_k8s_repo_infra//hack:verify-golangci-lint
-----------------------------------------------------------------------------
Verifying golangci-lint...
level=warning msg="[runner] Can't run linter goanalysis_metalinter: S1005: failed prerequisites: inspect@k8s.io/release/cmd/blocking-testgrid-tests, isgenerated@k8s.io/release/cmd/blocking-testgrid-tests"
level=warning msg="[runner] Can't run linter unused: buildssa: analysis skipped: errors in package: [/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/sandbox/processwrapper-sandbox/212/execroot/io_k8s_release/bazel-out/k8-fastbuild/bin/external/io_k8s_repo_infra/hack/verify-golangci-lint.runfiles/io_k8s_release/cmd/blocking-testgrid-tests/main.go:24:2: could not import net/http (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/net/http/client.go:13:2: could not import crypto/tls (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/crypto/tls/cipher_suites.go:16:2: could not import crypto/x509 (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/crypto/x509/verify.go:11:2: could not import net (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/net/cgo_linux.go:12:8: could not import C (cgo preprocessing failed))))) /root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/sandbox/processwrapper-sandbox/212/execroot/io_k8s_release/bazel-out/k8-fastbuild/bin/external/io_k8s_repo_infra/hack/verify-golangci-lint.runfiles/io_k8s_release/cmd/blocking-testgrid-tests/main.go:29:2: could not import k8s.io/test-infra/testgrid/config (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/sandbox/processwrapper-sandbox/212/execroot/io_k8s_release/_tmp/ca3cea22f615a4ff022845a3169046b1/home/go/pkg/mod/k8s.io/test-infra@v0.0.0-20190829230513-7ef687d80d22/testgrid/config/config.go:27:2: could not import cloud.google.com/go/storage (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/sandbox/processwrapper-sandbox/212/execroot/io_k8s_release/_tmp/ca3cea22f615a4ff022845a3169046b1/home/go/pkg/mod/cloud.google.com/go@v0.38.0/storage/acl.go:19:2: could not import net/http (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/net/http/client.go:13:2: could not import crypto/tls (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/crypto/tls/cipher_suites.go:16:2: could not import crypto/x509 (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/crypto/x509/verify.go:11:2: could not import net (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/net/cgo_linux.go:12:8: could not import C (cgo preprocessing failed)))))))]"
```

So yes, the linter fails to compile the package because of the missing c compiler. I would also prefer to add a dedicated build image from our side but I'm not sure where to host it.